charuco_detector:
  ros__parameters:
    charuco_parameters:
      squaresSidesSizeM: 0.0200
      markersSidesSizeM: 0.0150
      numberOfBitsForMarkersSides: 4
      numberOfMarkers: 70
      numberOfSquaresInX: 10
      numberOfSquaresInY: 14
      dictionaryId: 3
      
      # Detection parameters
      adaptiveThreshWinSizeMin: 3
      adaptiveThreshWinSizeMax: 23
      adaptiveThreshWinSizeStep: 10
      adaptiveThreshConstant: 7.0
      minMarkerPerimeterRate: 0.03
      maxMarkerPerimeterRate: 4.0
      polygonalApproxAccuracyRate: 0.03
      minCornerDistanceRate: 0.05
      minDistanceToBorder: 3
      minMarkerDistanceRate: 0.05
      cornerRefinementWinSize: 5
      cornerRefinementMaxIterations: 30
      cornerRefinementMinAccuracy: 0.1
      markerBorderBits: 1
      perspectiveRemovePixelPerCell: 4
      perspectiveRemoveIgnoredMarginPerCell: 0.13
      maxErroneousBitsInBorderRate: 0.35
      minOtsuStdDev: 5.0
      errorCorrectionRate: 0.6

    function_parameters:
      use_median_blur: false
      median_blur_k_size: 3
      use_dynamic_range: false
      use_bilateral_filter: false
      bilateral_filter_pixel_neighborhood: 5
      bilateral_filter_sigma_color: 100.0
      bilateral_filter_sigma_space: 100.0
      bilateral_filter_border_type: 4
      use_clahe: false
      clahe_clip_limit: 2.0
      clahe_size_x: 8
      clahe_size_y: 8
      use_adaptive_threshold: false
      adaptive_threshold_max_value: 255.0
      adaptive_threshold_method: 1
      adaptive_threshold_type: 0
      adaptive_threshold_block_size: 65
      adaptive_threshold_constant_offset_from_mean: 0.0

    tf_parameters:
      use_static_tf_broadcaster: false
      tf_broadcaster_republish_rate: 10.0
      sensor_frame_override: "camera_color_optical_frame"
      charuco_tf_frame: "charuco"

    subscribers:
      image_topic:
        topic: "/camera/color/image_raw"
      camera_info:
        topic: "/camera/color/camera_info"

    publishers:
      image_results_publisher:
        queue_size: 1
        latch: false
